lib/instrumentation/http-shared.js

```diff
@@ -164,6 +168,9 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
       if (getSafeHost(req) === agent._conf.serverHost) {
         agent.logger.debug('ignore %s request to intake API %o', moduleName, { id: id })
         return req
+      } else if (getInternalIgnoreHeader(req) === '1') {
+        agent.logger.debug('ignore %s request with ignore header flag %o', moduleName, { id: id })
+        return req
```

Comment:
This is basically a new feature inside this feature, it would allow us to ignore some http requests if they contain a specific header. I don't know if this header should be configured by the client or not, as it's being used only internally for now.